src/member-api.js

```diff
@@ -51,19 +52,34 @@ export class MemberApi extends TypedEmitter {
    *
    * @param {Object} opts
    * @param {import('./capabilities.js').RoleId} opts.roleId
+   * @param {string} [opts.roleDescription]
    * @param {number} [opts.timeout]
    *
    * @returns {Promise<import('./generated/rpc.js').InviteResponse_Decision>}
    */
-  async invite(deviceId, { roleId, timeout }) {
-    const projectId = projectKeyToId(this.#projectKey)
+  async invite(deviceId, { roleId, roleDescription, timeout }) {
+    const roleName = DEFAULT_CAPABILITIES[roleId].name
+    const { name: deviceName } = await this.getById(this.#ownDeviceId)
+
+    // since we are always getting #ownDeviceId,
+    // this should never throw (see comment on getById), but it pleases ts
+    if (!deviceName) throw new Error('Invalid deviceName')
+
+    if (!roleName) throw new Error('Invalid roleId')
 
+    const projectId = projectKeyToId(this.#projectKey)
     const project = await this.#dataTypes.project.getByDocId(projectId)
 
+    if (!project.name)
+      throw new Error('Project must have a name to invite people')
+
     const response = await this.#rpc.invite(deviceId, {
       projectKey: this.#projectKey,
       encryptionKeys: this.#encryptionKeys,
       projectInfo: { name: project.name },
+      roleName,
```

Comment:
would move the check on roleId as the first thing to do (small perf improvement). suggestion also incorporates specifying the roleName as a param as mentioned in my other comment

```suggestion
  async invite(deviceId, { roleId, roleName, roleDescription, timeout }) {
    if (!(roleId in DEFAULT_CAPABILITIES)) throw new Error('Unknown roleId')

    const { name: deviceName } = await this.getById(this.#ownDeviceId)

    // since we are always getting #ownDeviceId,
    // this should never throw (see comment on getById), but it pleases ts
    if (!deviceName) throw new Error('Invalid deviceName')

    const projectId = projectKeyToId(this.#projectKey)
    const project = await this.#dataTypes.project.getByDocId(projectId)

    if (!project.name)
      throw new Error('Project must have a name to invite people')

    const response = await this.#rpc.invite(deviceId, {
      projectKey: this.#projectKey,
      encryptionKeys: this.#encryptionKeys,
      projectInfo: { name: project.name },
      roleName: roleName || DEFAULT_CAPABILITIES[roleId].name,
```